pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -100,10 +100,18 @@ func (service *SecurityPolicyService) CreateOrUpdateSecurityPolicy(obj *v1alpha1
 	}
 
 	existingSecurityPolicy := service.securityPolicyStore.GetByKey(*nsxSecurityPolicy.Id)
-	existingRules := service.ruleStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
-	existingGroups := service.groupStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
-
+	tmpExistingRules := service.ruleStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
+	tmpExistingGroups := service.groupStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
+	existingRules := []model.Rule{}
```

Comment:
L103~L111 looks a little strange... Can we directly change `func RulesToComparable(rules []model.Rule) []Comparable {...}` to `func RulesToComparable(rules []*model.Rule) []Comparable {...}` then directly pass to list of pointers to the function `RulesToComparable`? Then we don't need to convert the list of pointers to the list of objects before passing the list of objects to `RulesToComparable`...

Reply:
Done